majilove/models.py

```diff
@@ -492,3 +493,205 @@ def __unicode__(self):
                 name=self.t_name,
             )
         return u''
+
+############################################################
+# Collectible Photos
+
+class CollectiblePhoto(AccountAsOwnerModel):
+    collection_name = 'collectiblephoto'
+
+    account = models.ForeignKey(Account, verbose_name=_('Account'), related_name='photoscollectors')
+    photo = models.ForeignKey(Photo, verbose_name=_('Photo'), related_name='collectedphotos')
+    level = models.PositiveIntegerField(_('Level'), default=1)
+    leader_bonus = models.PositiveIntegerField(_('Leader skill percentage'), null=True)
+    skill_level = models.PositiveIntegerField(_('Skill level'), default=1)
+
+    @property
+    def skill_percentage(self):
+        return self.photo.skill_percentage + (self.skill_level - 1) * self.photo.skill_increment
+    @property
+    def skill_note_count(self):
+        return self.photo.skill_note_count + (self.skill_level - 1) * self.photo.skill_increment
+
+    @property
+    def skill(self):
+        return self.photo.skill_template.format({
+            k: getattr(self, k)
+            for k in templateVariables(self.photo.skill_template)
+        })
+
+    sub_skill_level = models.PositiveIntegerField(_('Sub skill level'), null=True)
+    @property
+    def sub_skill_amount(self):
+        return self.photo.sub_skill_amount + (self.sub_skill_level - 1) * self.photo.sub_skill_increment
+
+    @property
+    def sub_skill(self):
+        _sub_skill_variables = {k: getattr(self.photo, k)
+        for k in templateVariables(self.photo.sub_skill_template
+        )}
+        _sub_skill_variables['sub_skill_amount'] = self.sub_skill_amount
+        return self.photo.sub_skill_template.format(**_sub_skill_variables)
+
+    rank = models.PositiveIntegerField(_('Rank'), default=1)
+
+    # TODO: moment based things are the same across, do I need an intermediate model for moment things?
```

Comment:
Not sure what you mean by that, can you tell me which models share the data?

Reply:
I also don't know what this means...